src/panels/developer-tools/service/developer-tools-service.ts

```diff
@@ -195,6 +199,15 @@ class HaPanelDevService extends LitElement {
     if (!serviceData?.service) {
       return false;
     }
+
+    this._error = undefined;
+    if (hasTemplate(this._serviceData!)) {
+      this._error = this.hass.localize(
```

Comment:
A memoize should not have side effects, you could return the error:
`return {isValid: false, error: error};`
and it render:
`const {isValid, error} = this._isValid(this._serviceData, fields, target);`